Weekly_Work/Week2/Metrics_Root_class.py

```diff
@@ -0,0 +1,98 @@
+import json
+import datetime
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+class Metric():
+    
+    def __init__(self, path, source_code_exclude_list=None):
+
+        self.source_code_exclude_list = source_code_exclude_list
+        self.repo_urls = set()
+
+        data_list = list()
+        self.raw_df = pd.DataFrame()
+
+        with open(path, 'r') as raw_data:
```

Comment:
I would rather do this (reading the list from the file) in an auxiliary function, so that the class receives the list when initializing.

Reply:
@Polaris000 you can add functions like these (since this function can be used in multiple places) in the utils.py file and use them from there

Reply:
Should I create a class for that? I think it should be left as a file with functions .

Reply:
A file that contains all these functions will be fine. Since these functions are not connected to each other and just used on a need bases, creating a Class to access these functions will not make much sense.

Remember, classes usually represent an entity with the variables representing the attributes of that entity and the functions of that class representing the behaviour of that entity (functions work on the attributes, probably modify them). Which is not required here

Reply:
Of course... Thanks a lot!

Reply:
Yes, I agree with @aswanipranjal. Maybe at some point later we see some benefit in using a class instead of a function, but for now, i would rather use a function.